### PR TITLE
test(scale): cover more operators in nexmark chaos tests

### LINKED ISSUE #7788: test: scaling of 2-phase aggregation is not covered
According to [nexmark_source.yaml](https://github.com/risingwavelabs/risingwave/blob/93a8e2c7205452b8252d5adbf01c0d7ce2dada72/src/frontend/planner_test/tests/testdata/nexmark_source.yaml#L1), there's no 2-phase aggregation (but 2-phase top-n) in the current chaos-test Nexmark queries. We should make up a new query to cover it.

- #6485

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR adds some tests to cover more operators in nexmark chaos tests.
- ~Add watermark column for nexmark-chaos and `nexmark_source.yaml`, to cover `WatermarkFilter`.~ Watermark does not survive scaling test, will investigate later.
- q15: Covers append-only hash aggregation with `distinct`.
- q18: Covers group top-n (possibly deduplication in the future).
- q106: self-made, covers two-phase stateful simple aggregation. (Close #7788)

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by lmatz — APPROVED** [review]
LGTM!

**Comment by BugenZhao** [comment]
Let's wait for #7623 getting merged and also add these queries to recovery test.

**Comment by wangrunji0408** [comment]
> Let's wait for #7623 getting merged and also add these queries to recovery test.

Just found a new bug from this PR. I've retried and I am not sure if this run is lucky enough to let it merged. 🤣

File: src/tests/simulation/src/nexmark.rs [hunk 1]
```diff
@@ -37,11 +37,14 @@ impl NexmarkCluster {
         conf: Configuration,
         split_num: usize,
         event_num: Option<usize>,
+        watermark: bool,
```
Comment (BugenZhao) [inline src/tests/simulation/src/nexmark.rs:40]: Watermark operators do not survive the scaling test, so I've made this always `false` now. Will investigate it later.

**Review by BugenZhao — COMMENTED** [review]
😟

**Pull request merged**